File: VIQA/HtmlElements/SimpleElements/Cell.cs
using VIQA.HtmlElements.Interfaces;

namespace VIQA.HtmlElements.SimpleElements
{
    public class Cell<T> : VIElement, IHaveValue where T : IHaveValue
    {
        public readonly T Element;
        public string Value { get { return Element.Value; } }
        public int ColumnNum;
        public int RowNum;
        public string ColumnName;
        public string RowName;

        public void SetValue<T1>(T1 value) { Element.SetValue(value); }

        public Cell(T element, int columnNum, int rowNum, string colName, string rowName)
        {
            Element = element;
            ColumnNum = columnNum;
            RowNum = rowNum;
            ColumnName = colName;
            RowName = rowName;
        }

        public Cell<T> UpdateData(string colName, string rowName)
        {
            if (string.IsNullOrEmpty(ColumnName) && !string.IsNullOrEmpty(colName))
                ColumnName = colName;
            if (string.IsNullOrEmpty(RowName) && !string.IsNullOrEmpty(rowName))
                RowName = rowName;
            return this;
        }
    }
}

File: Java/src/main/java/ru/viqa/ui_testing/elements/baseClasses/Table/Cell.java
package ru.viqa.ui_testing.elements.baseClasses.Table;

/**
 * Created by 12345 on 25.10.2014.
 */

import ru.viqa.ui_testing.elements.baseClasses.*;
import ru.viqa.ui_testing.elements.interfaces.IHaveValue;

public class Cell<T extends IHaveValue> extends VIElement {
    public T element;
    public int columnNum;
    public int rowNum;
    public String columnName;
    public String rowName;

    public Cell(T element, int columnNum, int rowNum, String colName, String rowName) throws Exception {
        this.element = element;
        this.columnNum = columnNum;
        this.rowNum = rowNum;
        columnName = colName;
        this.rowName = rowName;
    }

    public Cell<T> updateData(String colName, String rowName) {
        if ((columnName == null || columnName.equals("")) && !(colName == null || colName.equals("")))
            columnName = colName;
        if ((this.rowName == null || this.rowName.equals("")) && !(rowName == null || rowName.equals("")))
        this.rowName = rowName;
        return this;
    }
}
